tests/unit/test_authentication.py

```diff
@@ -88,6 +91,17 @@ def test_get_expiration_time_from_header(auth_header: str, expected_expiration:
     assert expiration == expected_expiration
 
 
+@pytest.mark.parametrize(
+    "auth_header",
+    (f"Bearer {jwt.encode({'exp': 1697013997}, 'secret')}XX", "Bearer NotAToken"),
+)
+def test_get_expiration_time_from_header_malformed_bearer_token_raises_unauthorized(auth_header: str) -> None:
```

Comment:
praise: Structuring tests this way makes it very easy to review and understand what the expected outcome is 👍